tsfc/loopy.py

```diff
@@ -190,6 +193,73 @@ def statement_evaluate(leaf, ctx):
         return ops
     elif isinstance(expr, gem.Constant):
         return []
+    elif isinstance(expr, gem.ComponentTensor):
+        idx = tuple()
+        sub_idx = ()
+        var = ctx.pymbolic_variable(expr)
+        for c, index in enumerate(expr.multiindex):
+            ctx.active_indices[index] = p.Variable(index.name)
+            ctx.index_extent[index.name] = index.extent
+            idx = idx + (ctx.active_indices[index], )
+        if isinstance(var, p.Subscript):
+            var, sub_idx = var.aggregate, var.index_tuple
+            var = p.Subscript(var, idx + sub_idx)
+            rhs = expression(expr.children[0], ctx)
+        else:
+            var = p.Subscript(var, idx)
+            rhs = expression(expr.children[0], ctx)
+        statements = [lp.Assignment(var, rhs, within_inames=ctx.active_inames())]
+        for c, index in enumerate(expr.multiindex):
+            ctx.active_indices.pop(index)
+        return statements
+    elif isinstance(expr, gem.Inverse):
+        idx = tuple()
+        idx2 = tuple()
+        for c, index in enumerate(expr.multiindex):
+            ctx.active_indices[index] = p.Variable(index.name)
+            ctx.index_extent[index.name] = index.extent
+            idx = idx + (ctx.active_indices[index], )
```

Comment:
Ah, so you use that multiindex for code generation...

Reply:
Yes, I use the indices (i0,i1) of the thing the Inverse acts on (t0) to feed it it into the inverse function call in loopy. The multiindices (i2,i3) of the gem Inverse node are for the left hand side (t1). So `t1[i2,i3]=inverse(t0[i0,i1])` would be the statement in loopy. When the loopy code is translated to, however, `t1[i2,i3]` gets dropped since the inverse is void and `t0` is in/out parameter. If I find a way to call the inverse without assigning it to an lhs in loopy, we could drop the multi index in the gem nodes.